### PR TITLE
Generating credentials using WebhookIdentity on AWS

### LINKED ISSUE #462: Need Support for IRSA on AWS
**Is your feature request related to a problem? Please describe.**
Currently the AWS signing works only when the request goes through EC2 metadata endpoint. We need to be able to auth using the Webhookidentity auth provided by AWS
https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

**Describe the solution you'd like**
Allow creating AWS session object using the webhook identity

**Describe alternatives you've considered**
Using KIAM or directly the EC2 instance is the only option which cuts down the principle of "least access"

**Additional context**
This could help us cutting down the dependency on Kiam and rather use native AWS provided kubernetes JWT signing

### PR DESCRIPTION
**Pull request by kartik-moolya**
##### ISSUE TYPE

 - Feature Pull Request

##### SUMMARY

Capability to fetch AWS credentials from the EKS OIDC provider

Fixes #462

### CODE REVIEW

File: pkg/notify/elasticsearch.go [hunk 1]
```diff
@@ -61,7 +62,13 @@ func NewElasticSearch(c config.ElasticSearch) (Notifier, error) {
 	if c.AWSSigning.Enabled {
 		// Get credentials from environment variables and create the AWS Signature Version 4 signer
 		sess := session.Must(session.NewSession())
-		if c.AWSSigning.RoleArn != "" {
+
+		// Use OIDC token to generate credentials if using IAM to Service Account
+		awsRoleARN := os.Getenv("AWS_ROLE_ARN")
+		awsWebIdentityTokenFile := os.Getenv("AWS_WEB_IDENTITY_TOKEN_FILE")
```
Comment (PrasadG193) [inline pkg/notify/elasticsearch.go:68]: Please declare "AWS_ROLE_ARN" and "AWS_WEB_IDENTITY_TOKEN_FILE" as consts

Comment (kartik-moolya) [reply pkg/notify/elasticsearch.go:68]: These are the environment variables which will be injected to the pod when we use the eks webhook identity annotation.

Comment (kartik-moolya) [reply pkg/notify/elasticsearch.go]: Done

File: helm/botkube/README.md [hunk 2]
```diff
@@ -17,6 +17,18 @@ Controller for the BotKube Slack app which helps you monitor your Kubernetes clu
 
 * <https://github.com/infracloudio/botkube>
 
+### Now Supports AWS IRSA on EKS
+
+AWS has introduced IAM Role for Service Accounts in order to provide fine grained access. This is useful if you are looking to run botkube inside an EKS cluster. For more details visit https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html.
```
Comment (PrasadG193) [inline helm/botkube/README.md:22]:
```suggestion
AWS has introduced IAM Role for Service Accounts in order to provide fine grained access. This is useful if you are looking to run BotKube inside an EKS cluster. For more details visit https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html.
```

File: helm/botkube/README.md [hunk 3]
```diff
@@ -17,6 +17,18 @@ Controller for the BotKube Slack app which helps you monitor your Kubernetes clu
...
+
+Annotate the Botkube Service Account as shown in the example below and add the necessary Trust Relationship to the corresponding Botkube role to get this working
```
Comment (PrasadG193) [inline helm/botkube/README.md:24]:
```suggestion
Annotate the BotKube Service Account as shown in the example below and add the necessary Trust Relationship to the corresponding BotKube role to get this working
```

**Review by PrasadG193 — APPROVED** [review]

**Pull request merged**